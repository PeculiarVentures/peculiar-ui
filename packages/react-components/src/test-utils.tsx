import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
