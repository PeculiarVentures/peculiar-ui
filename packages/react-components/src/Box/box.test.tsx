import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { Box } from './index';

describe('<Box />', () => {
  it('should render with default styles', () => {
    const { asFragment } = render(<Box />);

    expect(asFragment()).toMatchSnapshot();
  });
});
