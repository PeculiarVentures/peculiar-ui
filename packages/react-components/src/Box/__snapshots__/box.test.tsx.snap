// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Box /> > border positions > position "bottom" 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-color: var(--pv-color-attention);
  border-style: solid;
  border-radius: 4px;
  border-width: 0px 0px 1px 0px;
}

<div
    class="emotion-0"
  >
    Message
  </div>
</DocumentFragment>
`;

exports[`<Box /> > border positions > position "horizontal" 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-color: var(--pv-color-attention);
  border-style: solid;
  border-radius: 4px;
  border-width: 1px 0px 1px 0px;
}

<div
    class="emotion-0"
  >
    Message
  </div>
</DocumentFragment>
`;

exports[`<Box /> > border positions > position "left" 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-color: var(--pv-color-attention);
  border-style: solid;
  border-radius: 4px;
  border-width: 0px 0px 0px 1px;
}

<div
    class="emotion-0"
  >
    Message
  </div>
</DocumentFragment>
`;

exports[`<Box /> > border positions > position "right" 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-color: var(--pv-color-attention);
  border-style: solid;
  border-radius: 4px;
  border-width: 0px 1px 0px 0px;
}

<div
    class="emotion-0"
  >
    Message
  </div>
</DocumentFragment>
`;

exports[`<Box /> > border positions > position "top" 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-color: var(--pv-color-attention);
  border-style: solid;
  border-radius: 4px;
  border-width: 1px 0px 0px 0px;
}

<div
    class="emotion-0"
  >
    Message
  </div>
</DocumentFragment>
`;

exports[`<Box /> > border positions > position "vertical" 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-color: var(--pv-color-attention);
  border-style: solid;
  border-radius: 4px;
  border-width: 0px 1px 0px 1px;
}

<div
    class="emotion-0"
  >
    Message
  </div>
</DocumentFragment>
`;

exports[`<Box /> > box shadow > shadow "dark-hight" 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-shadow: var(--pv-shadow-dark-hight);
}

<div
    class="emotion-0"
  >
    Message
  </div>
</DocumentFragment>
`;

exports[`<Box /> > box shadow > shadow "dark-medium" 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-shadow: var(--pv-shadow-dark-medium);
}

<div
    class="emotion-0"
  >
    Message
  </div>
</DocumentFragment>
`;

exports[`<Box /> > box shadow > shadow "light-hight" 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-shadow: var(--pv-shadow-light-hight);
}

<div
    class="emotion-0"
  >
    Message
  </div>
</DocumentFragment>
`;

exports[`<Box /> > box shadow > shadow "light-low" 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-shadow: var(--pv-shadow-light-low);
}

<div
    class="emotion-0"
  >
    Message
  </div>
</DocumentFragment>
`;

exports[`<Box /> > box shadow > shadow "light-medium" 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-shadow: var(--pv-shadow-light-medium);
}

<div
    class="emotion-0"
  >
    Message
  </div>
</DocumentFragment>
`;

exports[`<Box /> > box shadow > shadow "light-soft" 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-shadow: var(--pv-shadow-light-soft);
}

<div
    class="emotion-0"
  >
    Message
  </div>
</DocumentFragment>
`;

exports[`<Box /> > should pass className 1`] = `
<DocumentFragment>
  <div
    class="my-class-name emotion-0"
  />
</DocumentFragment>
`;

exports[`<Box /> > should render with default styles 1`] = `
<DocumentFragment>
  <div
    class="emotion-0"
  />
</DocumentFragment>
`;
